include/crocoddyl/multibody/contacts/contact-6d-loop.hxx

```diff
@@ -0,0 +1,261 @@
+      pinocchio::LOCAL, d->v2_partial_dq, d->a2_partial_dq, d->a2_partial_dv,
+      d->a2_partial_da);
+
+  d->da0_dq_t1 =
+      joint1_placement_.toActionMatrixInverse() * d->a1_partial_dq;  //
+  d->da0_dq_t2 = (d->f1af2.toActionMatrix() * (d->f1Jf1 - d->f1Xf2 * d->f2Jf2) +
+                  d->f1Xf2 * (joint2_placement_.toActionMatrixInverse() *
+                              d->a2_partial_dq));  //
+  d->da0_dq_t3 =
+      -d->f1vf2.toActionMatrix() * (joint1_placement_.toActionMatrixInverse() *
+                                    d->v1_partial_dq)  // part 1
+      + d->f1vf1.toActionMatrix() * d->f1vf2.toActionMatrix() *
+            (d->f1Jf1 - d->f1Xf2 * d->f2Jf2)  // part 2
+      + d->f1vf1.toActionMatrix() * d->f1Xf2 *
+            (joint2_placement_.toActionMatrixInverse() *
+             d->v2_partial_dq);  // part 3
+
+  d->da0_dx.leftCols(nv) = d->da0_dq_t1 - d->da0_dq_t2 + d->da0_dq_t3;
+  d->da0_dx.rightCols(nv) =
+      joint1_placement_.toActionMatrixInverse() * d->a1_partial_dv -
+      d->f1Xf2 *
+          (joint2_placement_.toActionMatrixInverse() * d->a2_partial_dv) -
+      d->f1vf2.toActionMatrix() * d->f1Jf1 +
+      d->f1vf1.toActionMatrix() * d->f1Xf2 * d->f2Jf2;
+  if (gains_[0] != 0.) {
+    Matrix6s f1Mf2_log6;
+    pinocchio::Jlog6(d->f1Mf2, f1Mf2_log6);
+    d->da0_dx.leftCols(nv) +=
+        gains_[0] * (-f1Mf2_log6 * (-d->oMf2.toActionMatrixInverse() *
+                                        d->oMf1.toActionMatrix() * d->f1Jf1 +
+                                    d->f2Jf2));
+  }
+  if (gains_[1] != 0.) {
+    d->da0_dx.leftCols(nv) +=
+        gains_[1] *
+        (joint1_placement_.toActionMatrixInverse() * d->v1_partial_dq -
+         d->f1Mf2.act(d->f2vf2).toActionMatrix() *
+             (d->f1Jf1 - d->f1Xf2 * d->f2Jf2) -
+         d->f1Xf2 * joint2_placement_.toActionMatrixInverse() *
+             d->v2_partial_dq);
+    d->da0_dx.rightCols(nv) += gains_[1] * (d->f1Jf1 - d->f1Xf2 * d->f2Jf2);
+  }
+}
+
+template <typename Scalar>
+void ContactModel6DLoopTpl<Scalar>::updateForce(
+    const boost::shared_ptr<ContactDataAbstract> &data, const VectorXs &force) {
+  if (force.size() != 6) {
+    throw_pretty(
+        "Invalid argument: " << "lambda has wrong dimension (it should be 6)");
+  }
+  Data *d = static_cast<Data *>(data.get());
+  d->f = pinocchio::ForceTpl<Scalar>(-force);
+  switch (type_) {
+    case pinocchio::ReferenceFrame::LOCAL: {
+      data->fext = joint1_placement_.act(data->f);
+      d->joint1_f = -joint1_placement_.act(data->f);
+      d->joint2_f = (joint2_placement_ * d->f1Mf2.inverse()).act(data->f);
+
+      data->dtau_dq.setZero();
+
+      d->f_cross.setZero();
+      d->f_cross.topRightCorner(3, 3) = pinocchio::skew(d->joint2_f.linear());
+      d->f_cross.bottomLeftCorner(3, 3) = pinocchio::skew(d->joint2_f.linear());
+      d->f_cross.bottomRightCorner(3, 3) =
+          pinocchio::skew(d->joint2_f.angular());
```

Comment:
```suggestion
      Matrix6s f_cross = Matrix6s::Zero(6, 6);
      f_cross.topRightCorner(3, 3) = pinocchio::skew(d->joint2_f.linear());
      f_cross.bottomLeftCorner(3, 3) = pinocchio::skew(d->joint2_f.linear());
      f_cross.bottomRightCorner(3, 3) = pinocchio::skew(d->joint2_f.angular());
```
As this matrix exists only in this place, I would opt for using a local variable for it.